pypesto/visualize/misc.py

```diff
@@ -80,18 +80,16 @@ def process_result_list(
                 legends.append('Result ' + str(i_leg))
         else:
             # legends were passed by user: check length
-            if isinstance(legends, list):
+            try:
+                if isinstance(legends, str):
+                    legends = [legends]
```

Comment:
This checks if legends is a string (returns False in case of tuple). If it is it will create a list of length 1 which should always throw an error then.

to catch tuples and and other sequence types, perhaps rather use `legends  list(legends)`.
```suggestion
                legends = list(legends)
```

Reply:
The idea was that a string of length `!= len(results)` would throw an error, but a string with the appropriate number of characters would be used for the legend. We could also raise an error for any string, but this would not be consistent with the type hint for the legend argument of `visualize.waterfall`: `legends: Optional[Union[Sequence[str], str]] = None,` because a string is a Sequence.

I don't know what the second suggestion is. A tuple works now, as it should. Any datatype for which the `length` function is not defined raises a `TypeError`.

Reply:
ahhh understood, thanks. Was a bit unclear for me.